main.go

```diff
@@ -111,3 +120,31 @@ func run() error {
 		time.Sleep(5 * time.Second)
 	}
 }
+
+func setupKubeconfigWatches(o *options) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return fmt.Errorf("failed to set up watcher: %w", err)
+	}
+	for _, candidate := range []string{o.BuildClusterKubeconfig, o.ReleaseClusterKubeconfig, "/var/run/secrets/kubernetes.io/serviceaccount/token"} {
+		if _, err := os.Stat(candidate); err != nil {
+			continue
+		}
+		if err := watcher.Add(candidate); err != nil {
+			return fmt.Errorf("failed to watch %s: %w", candidate, err)
+		}
+	}
+
+	go func() {
+		for e := range watcher.Events {
+			if e.Op == fsnotify.Chmod {
+				// For some reason we get frequent chmod events from Openshift
+				continue
+			}
+			klog.Infof("event: %s, kubeconfig changed, exiting to make the kubelet restart us so we can pick them up", e.String())
+			interrupts.Terminate()
```

Comment:
Maybe just directly `os.Exit(0)`, I don't think indirecting over `interrupts` when the package is otherwise unused will add anything useful?

Reply:
Makes sense.  I have removed `interrupts` all together.